content/hello-world-program.md

```diff
@@ -0,0 +1,295 @@
+
+```rust
+use solana_program::account_info::AccountInfo
+```
+
+For a basic program we will need the following:
+
+```rust
+use solana_program::{
+  account_info::AccountInfo,
+  entrypoint,
+  entrypoint::ProgramResult,
+  pubkey::Pubkey,
+  msg
+};
+```
+
+Solana programs require a single entry point to process program instructions. The entry point is declared using the [entrypoint!](https://docs.rs/solana-program/latest/solana_program/macro.entrypoint.html) macro. You can read more about Rust macros [here](https://doc.rust-lang.org/book/ch19-06-macros.html).
+
+The entry point to a Solana program requires a `process_instruction` function with the following arguments: 
+
+- `program_id` is the address of the account the program is stored
+- `accounts` is the list of accounts required to process the instruction
+- `instruction_data` is the serialized instruction-specific data
+
+```rust
+entrypoint!(process_instruction);
+
+fn process_instruction(
+    program_id: &Pubkey,      
+    accounts: &[AccountInfo], 
+    instruction_data: &[u8], 
+) -> ProgramResult;
+```
+
+Recall that Solana programs accounts only store instruction logic. This means programs are read-only and “stateless”. The “state” (data) that a program requires to process an instruction are stored in data accounts. 
+
+In order to process an instruction, the data accounts that an instruction requires must be explicitly passed into the program through the `accounts` argument. Any additional inputs must be passed in through the `instruction_data` argument.
+
+### Transactions
+
+In order to invoke Solana program instructions, clients must build and submit a transaction. Recall that transactions are made up of one more or instructions. Transactions process instructions in order and “atomically”. Atomically means that if any instruction within the transaction fails, then the entire transaction will fail. This provides users with the certainty that if a transaction is confirmed, then all instructions within the transaction were processed successfully. 
+
+Building a basic transaction requires the following steps: 
+
+1. Create a new transaction
+2. Create an instruction
+3. Add the instruction to the transaction
+4. Submit the transaction to be processed by the Solana network 
+
+First, we must create a new transaction using the `Transaction` function from `@solana/web3.js`.
+
+```tsx
+import web3 = require("@solana/web3.js");
+
+const transaction = new web3.Transaction()
+```
+
+Next, we build the instruction using the `TransactionInstruction` function. The `keys` field is where the accounts required by the instruction are listed. Note that the format for building a new instruction on the client side matches the arguments required by the `process_instruction` function passed into the program entry point.
```

Comment:
@ZYJLiu does the first account listed in the keys field array have any responsibilities relative to the first index?

i.e. like in the Intro to Writing Data lesson,
> The first signer included in the array of signers on a transaction is responsible for paying the transaction fee.
🤔

Reply:
I think that's referring to when a transaction is submitted from the client side

>     const sig = await web3.sendAndConfirmTransaction(
        connection,
        new web3.Transaction().add(instruction),
        [payer]
    )

Reply:
not sure if the order of accounts passed into keys has to have to "payer" as the first account

I think that part about the first signer included in the array refers to when a transaction is sent (an if there are multiple signers required)
 `const transactionSignature = await web3.sendAndConfirmTransaction(
    connection,
    transaction,
    [payer] <--- in case there are multiple signers listed here, the first one is the payer for the transaction
  );`

Reply:
Cool - thanks @ZYJLiu